core/src/main/java/org/elasticsearch/rest/action/admin/cluster/node/hotthreads/RestNodesHotThreadsAction.java

```diff
@@ -40,19 +38,17 @@
 /**
  */
 public class RestNodesHotThreadsAction extends BaseRestHandler {
+    public RestNodesHotThreadsAction(RestGlobalContext context) {
+        super(context);
+        context.getController().registerHandler(RestRequest.Method.GET, "/_cluster/nodes/hotthreads", this);
```

Comment:
can we maybe have a final registerHandler method on `BaseRestHandler`?  eventually we should do this also in the module but that's a lot of work